lib/internal/Magento/Framework/App/Response/Http/FileFactory.php

```diff
@@ -48,13 +48,44 @@ public function __construct(
      * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      * @SuppressWarnings(PHPMD.NPathComplexity)
      * @SuppressWarnings(PHPMD.ExitExpression)
+     *
+     * @deprecated
      */
     public function create(
         $fileName,
         $content,
         $baseDir = DirectoryList::ROOT,
         $contentType = 'application/octet-stream',
         $contentLength = null
+    ) {
+        return $this->createWithParameters($fileName, $content, $baseDir, $contentType, $contentLength);
+    }
+
+    /**
+     * Declare headers and content file in response for file download
+     *
+     * @param string $fileName
+     * @param string|array $content set to null to avoid starting output, $contentLength should be set explicitly in
+     *                              that case
+     * @param string $baseDir
+     * @param string $contentType
+     * @param int $contentLength explicit content length, if strlen($content) isn't applicable
+     * @param string $dispositionType defines where to show content e.g. attachment or inline
+     * @throws \Exception
+     * @throws \InvalidArgumentException
+     * @return \Magento\Framework\App\ResponseInterface
+     *
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     * @SuppressWarnings(PHPMD.ExitExpression)
+     */
+    public function createWithParameters(
```

Comment:
According to https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/#php it is prohibited to introduce a new public method (at least in the patch release). Please rework it according to the mentioned guide

Reply:
@slavvka 
My intention wasn't to deprecate the old method, but to add a new parameter to set the disposition type.
How are we supposed to call the new functionality then without rewriting/setting a new preference for the class then? Would it be better to merge the changes to 2.3?

Reply:
You're completely right. You need to make a new preference because 2.3.0 is already frozen for such code changes.